h/static/scripts/directive/test/annotation-test.js

```diff
@@ -0,0 +1,1377 @@
+        assert(controller.document.uri === $scope.annotation.uri);
+      }
+    );
+
+    it(
+      'still sets the domain correctly if the annotation has no document',
+      function() {
+        delete annotation.document;
+        controller.render();
+        assert(controller.document.domain === 'example.com');
+      }
+    );
+
+    it(
+      'uses the domain for the title when the annotation has no document',
+      function() {
+        delete annotation.document;
+        controller.render();
+        assert(controller.document.title === 'example.com');
+      }
+    );
+
+    describe('timestamp', function() {
+      var clock;
+
+      beforeEach(function() {
+        clock = sinon.useFakeTimers();
+        annotation.created = (new Date()).toString();
+        annotation.updated = (new Date()).toString();
+      });
+
+      afterEach(function() {
+        clock.restore();
+      });
+
+      it('is not updated for unsaved annotations', function() {
+        // Unsaved annotations don't have an updated time yet so a timestamp
+        // string can't be computed for them.
+        annotation.updated = null;
+        $scope.$digest();
+        assert.equal(controller.timestamp, null);
+      });
+
+      it('is updated on first digest', function() {
+        $scope.$digest();
+        assert.equal(controller.timestamp, 'a while ago');
+      });
+
+      it('is updated after a timeout', function() {
+        fakeTime.nextFuzzyUpdate.returns(10);
+        fakeTime.toFuzzyString.returns('ages ago');
+        $scope.$digest();
+        clock.tick(11000);
+        $timeout.flush();
+        assert.equal(controller.timestamp, 'ages ago');
+      });
+
+      it('is no longer updated after the scope is destroyed', function() {
+        $scope.$digest();
+        $scope.$destroy();
+        $timeout.flush();
+        $timeout.verifyNoPendingTasks();
+      });
+    });
+
+    describe('share', function() {
+      var dialog;
+
+      beforeEach(function() {
+        dialog = $element.find('.share-dialog-wrapper');
+      });
+
+      it('sets and unsets the open class on the share wrapper', function() {
+        dialog.find('button').click();
+        isolateScope.$digest();
+        assert.ok(dialog.hasClass('open'));
+        $document.click();
+        assert.notOk(dialog.hasClass('open'));
+      });
+    });
+  });
+
+  describe('deleteAnnotation() method', function() {
+    beforeEach(function() {
+      createDirective();
+      fakeAnnotationMapper.deleteAnnotation = sandbox.stub();
+      fakeFlash.error = sandbox.stub();
+    });
+
+    it(
+      'calls annotationMapper.delete() if the delete is confirmed',
+      function(done) {
+        sandbox.stub($window, 'confirm').returns(true);
+        fakeAnnotationMapper.deleteAnnotation.returns($q.resolve());
+        controller['delete']().then(function() {
+          assert(fakeAnnotationMapper.deleteAnnotation.calledWith(annotation));
+          done();
+        });
```

Comment:
Missing a `.catch()`  block here. Alternatively, return the promise from the function instead of using `done` and the test harness should handle this for you.